TTCN3_Project/CANFilter/com/testingtech/playits/canfilter/CANFilterService.java

```diff
@@ -0,0 +1,216 @@
+          .println("[CANFilterService] Usage: java CANFilterService "
+              + "<Server host>  <Server Port Number>");
+      System.err.println("  ... using default server host "
+          + DEFAULT_SERVER_HOST + " default server port "
+          + DEFAULT_SERVER_PORT);
+    } else {
+      host = args[0];
+      port = extractPortNumber(args, 1, DEFAULT_SERVER_PORT);
+    }
+
+    InetSocketAddress address = new InetSocketAddress(host, port);
+    Socket socket = null;
+    ServerSocket serverSocket = null;
+    try {
+      serverSocket = createServerSocket(address);
+      System.out.println("[CANFilterService] Listening on "
+              + serverSocket.getInetAddress().getHostName() + ":"
+              + serverSocket.getLocalPort());
+      socket = serverSocket.accept();
+      timeoutResponder = new TimeoutResponder(socket, car2xEntries);
+      JSONTokener jsonTokener = new JSONTokener(socket.getInputStream());
+      while (jsonTokener.more()) {
+        try {
+          Object nextValue = jsonTokener.nextValue();
+          if (nextValue instanceof JSONObject) {
+            System.out
+                .println("[CANFilterService] Incoming request: "
+                    + nextValue);
+            // TODO maybe add a kill request to exit loop
+            handleRequest((JSONObject) nextValue);
+          } else {
+            System.err
+                .println("[CANFilterService] Could not process json request."
+                    + nextValue);
+          }
+        } catch (JSONException e) {
+          System.err.println("[CANFilterService] Could not process"
+              + " request. " + e.getMessage());
+        }
+      }
+
+    } catch (IOException e) {
+      System.err.println("[CANFilterService] Error listening on port "
+          + address.getPort() + ": " + e);
+    } catch (JSONException e) {
+      System.err
+          .println("[CANFilterService] Error while reading json objects from input stream");
+    } finally {
+      System.out
+          .println("[CANFilterService] No more requests. Closing sockets");
+      close(socket);
+      close(serverSocket);
+    }
+  }
+
+  private static void close(ServerSocket serverSocket) {
+    try {
+      if (serverSocket != null) {
+        serverSocket.close();
+      }
+    } catch (IOException e) {
+      System.err.println("[CANFilterService] Could not close socket: "
+          + e.getMessage());
+    }
+  }
+
+  private static void close(Socket socket) {
+    try {
+      if (socket != null) {
+        socket.close();
+      }
+    } catch (IOException e) {
+      System.err.println("[CANFilterService] Could not close socket: "
+          + e.getMessage());
+    }
+  }
+
+  private static ServerSocket createServerSocket(InetSocketAddress address)
+      throws IOException, SocketException {
+    ServerSocket serverSocket = new ServerSocket();
+    serverSocket.setReuseAddress(true);
+    serverSocket.bind(address);
+    return serverSocket;
+  }
+
```

Comment:
The reading of the json file and saving the data should definitely happen in separate classes/methods.